fastval.go

```diff
@@ -170,13 +170,39 @@ func (val FastVal) AsJsonString() (FastVal, error) {
 	return val, errors.New("invalid type coercion")
 }
 
+func (val FastVal) AsFloat64() (float64, error) {
+	switch val.dataType {
+	case FloatValue:
+		return float64(val.GetFloat()), nil
+	case IntValue:
+		return float64(val.GetInt()), nil
+	case UintValue:
+		return float64(val.GetUint()), nil
+	}
+
+	return 0, errors.New("invalid type coercion")
+}
+
 func (val FastVal) compareStrings(other FastVal) int {
 	// TODO: Improve string comparisons to avoid casting or converting
 	escVal, _ := val.AsJsonString()
 	escOval, _ := other.AsJsonString()
 	return strings.Compare(string(escVal.sliceData), string(escOval.sliceData))
 }
 
+func (val FastVal) compareNumeric(other FastVal) int {
```

Comment:
this method assumes that val and other are numeric types. better add a comment to emphasize this

Reply:
Again, I think this portion of the PR will be superseeded by PR #11 which has a more exhaustive implementation of the various FastVal comparisons.